exporter/exporterparser/exparser.go

```diff
@@ -32,7 +32,32 @@ import (
 	tracetranslator "github.com/census-instrumentation/opencensus-service/translator/trace"
 )
 
-func exportSpans(ctx context.Context, exporterName string, te trace.Exporter, td data.TraceData) error {
+// ocProtoSpansToOCSpanDataInstrumented converts
+// OpenCensus Proto TraceData to OpenCensus-Go SpanData.
+// The "Instrumented" suffix serves to document that this
+// function is traced but also has stats for self-observability.
+//
+// This is a bootstrapping mechanism for us to re-use as many of
+// the OpenCensus-Go trace.SpanData exporters which were written
+// by various vendors and contributors. Eventually the goal is to
+// get those exporters converted to directly receive
+// OpenCensus Proto TraceData.
+func ocProtoSpansToOCSpanDataInstrumented(ctx context.Context, exporterName string, te trace.Exporter, td data.TraceData) (aerr error) {
+	ctx, span := trace.StartSpan(ctx,
+		"opencensus.service.exporter."+exporterName+".ExportTrace",
+		trace.WithSampler(trace.NeverSample()))
+
+	span.Annotate([]trace.Attribute{
+		trace.Int64Attribute("n_spans", int64(len(td.Spans))),
+	}, "")
+
+	defer func() {
+		if aerr != nil {
```

Comment:
fancy golang thingy :) for a bit I was thinking where is this being set ...

Reply:
Yahh digg!